### ISSUE TITLE
BUG: Invalid IV length

### ISSUE THREAD

**Issue by UtechtDustin** [beekeeper-studio/beekeeper-studio#333]
**Describe the bug**
I will get the following error for the newest commit/master branch.
I'm not able to figure out the issue, but it could be a issue with #322 

@rathboma 
Does it makes sense to create a community chat e.g. slack/discord etc. for smaller questions ?
```
vue.runtime.esm.js?2b0e:619 [Vue warn]: Error in mounted hook (Promise/async): "Error: Invalid IV length"

found in

---> <ConnectionInterface> at src/components/ConnectionInterface.vue
       <App> at src/App.vue
         <Root>
```

**Comment by UtechtDustin** [comment]
After deletion the beekeeper database it works, so it seems it's an issue with the encryption.
I'm not sure if this is a issue because of the branch hopping or not, this should be checked twice (otherwise the connections could be corrupted)

**Comment by rathboma** [comment]
Damn. Any errors in the terminal?

**Comment by rathboma** [comment]
Yeah don't want to corrupt connections at all

**Comment by UtechtDustin** [comment]
Errors from the dev tools:
```
vue.runtime.esm.js?2b0e:619 [Vue warn]: Error in mounted hook (Promise/async): "Error: Invalid IV length"

found in

---> <ConnectionInterface> at src/components/ConnectionInterface.vue
       <App> at src/App.vue
         <Root>
warn @ vue.runtime.esm.js?2b0e:619
logError @ vue.runtime.esm.js?2b0e:1884
globalHandleError @ vue.runtime.esm.js?2b0e:1879
handleError @ vue.runtime.esm.js?2b0e:1839
eval @ vue.runtime.esm.js?2b0e:1856
Promise.catch (async)
invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1856
callHook @ vue.runtime.esm.js?2b0e:4219
insert @ vue.runtime.esm.js?2b0e:3139
invokeInsertHook @ vue.runtime.esm.js?2b0e:6346
patch @ vue.runtime.esm.js?2b0e:6565
Vue._update @ vue.runtime.esm.js?2b0e:3945
updateComponent @ vue.runtime.esm.js?2b0e:4066
get @ vue.runtime.esm.js?2b0e:4479
Watcher @ vue.runtime.esm.js?2b0e:4468
mountComponent @ vue.runtime.esm.js?2b0e:4073
Vue.$mount @ vue.runtime.esm.js?2b0e:8415
_callee$ @ main.ts?bc82:94
tryCatch @ runtime.js?96cf:45
invoke @ runtime.js?96cf:274
prototype.<computed> @ runtime.js?96cf:97
asyncGeneratorStep @ asyncToGenerator.js?1da1:3
_next @ asyncToGenerator.js?1da1:25
Promise.then (async)
asyncGeneratorStep @ asyncToGenerator.js?1da1:13
_next @ asyncToGenerator.js?1da1:25
vue.runtime.esm.js?2b0e:1888 Error: Invalid IV length
    at Decipheriv.createCipherBase (internal/crypto/cipher.js:101)
    at Decipheriv.createCipherWithIV (internal/crypto/cipher.js:119)
    at new Decipheriv (internal/crypto/cipher.js:260)
    at Object.createDecipheriv (crypto.js:123)
    at decryptString (events.js?2889:53)
    at eval (events.js?2889:45)
    at eval (events.js?2889:21)
    at Array.forEach (<anonymous>)
    at forMatchingColumns (events.js?2889:19)
    at Object.exports.decrypt (events.js?2889:44)
logError @ vue.runtime.esm.js?2b0e:1888
globalHandleError @ vue.runtime.esm.js?2b0e:1879
handleError @ vue.runtime.esm.js?2b0e:1839
eval @ vue.runtime.esm.js?2b0e:1856
Promise.catch (async)
invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1856
callHook @ vue.runtime.esm.js?2b0e:4219
insert @ vue.runtime.esm.js?2b0e:3139
invokeInsertHook @ vue.runtime.esm.js?2b0e:6346
patch @ vue.runtime.esm.js?2b0e:6565
Vue._update @ vue.runtime.esm.js?2b0e:3945
updateComponent @ vue.runtime.esm.js?2b0e:4066
get @ vue.runtime.esm.js?2b0e:4479
Watcher @ vue.runtime.esm.js?2b0e:4468
mountComponent @ vue.runtime.esm.js?2b0e:4073
Vue.$mount @ vue.runtime.esm.js?2b0e:8415
_callee$ @ main.ts?bc82:94
tryCatch @ runtime.js?96cf:45
invoke @ runtime.js?96cf:274
prototype.<computed> @ runtime.js?96cf:97
asyncGeneratorStep @ asyncToGenerator.js?1da1:3
_next @ asyncToGenerator.js?1da1:25
Promise.then (async)
asyncGeneratorStep @ asyncToGenerator.js?1da1:13
_next @ asyncToGenerator.js?1da1:25



vue.runtime.esm.js?2b0e:619 [Vue warn]: Error in mounted hook (Promise/async): "Error: Invalid IV length"

found in

---> <ConnectionInterface> at src/components/ConnectionInterface.vue
       <App> at src/App.vue
         <Root>
warn @ vue.runtime.esm.js?2b0e:619
logError @ vue.runtime.esm.js?2b0e:1884
globalHandleError @ vue.runtime.esm.js?2b0e:1879
handleError @ vue.runtime.esm.js?2b0e:1839
eval @ vue.runtime.esm.js?2b0e:1856
Promise.catch (async)
invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1856
callHook @ vue.runtime.esm.js?2b0e:4219
insert @ vue.runtime.esm.js?2b0e:3139
invokeInsertHook @ vue.runtime.esm.js?2b0e:6346
patch @ vue.runtime.esm.js?2b0e:6565
Vue._update @ vue.runtime.esm.js?2b0e:3945
updateComponent @ vue.runtime.esm.js?2b0e:4066
get @ vue.runtime.esm.js?2b0e:4479
Watcher @ vue.runtime.esm.js?2b0e:4468
mountComponent @ vue.runtime.esm.js?2b0e:4073
Vue.$mount @ vue.runtime.esm.js?2b0e:8415
_callee$ @ main.ts?bc82:94
tryCatch @ runtime.js?96cf:45
invoke @ runtime.js?96cf:274
prototype.<computed> @ runtime.js?96cf:97
asyncGeneratorStep @ asyncToGenerator.js?1da1:3
_next @ asyncToGenerator.js?1da1:25
Promise.then (async)
asyncGeneratorStep @ asyncToGenerator.js?1da1:13
_next @ asyncToGenerator.js?1da1:25
vue.runtime.esm.js?2b0e:1888 Error: Invalid IV length
    at Decipheriv.createCipherBase (internal/crypto/cipher.js:101)
    at Decipheriv.createCipherWithIV (internal/crypto/cipher.js:119)
    at new Decipheriv (internal/crypto/cipher.js:260)
    at Object.createDecipheriv (crypto.js:123)
    at decryptString (events.js?2889:53)
    at eval (events.js?2889:45)
    at eval (events.js?2889:21)
    at Array.forEach (<anonymous>)
    at forMatchingColumns (events.js?2889:19)
    at Object.exports.decrypt (events.js?2889:44)
logError @ vue.runtime.esm.js?2b0e:1888
globalHandleError @ vue.runtime.esm.js?2b0e:1879
handleError @ vue.runtime.esm.js?2b0e:1839
eval @ vue.runtime.esm.js?2b0e:1856
Promise.catch (async)
invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1856
callHook @ vue.runtime.esm.js?2b0e:4219
insert @ vue.runtime.esm.js?2b0e:3139
invokeInsertHook @ vue.runtime.esm.js?2b0e:6346
patch @ vue.runtime.esm.js?2b0e:6565
Vue._update @ vue.runtime.esm.js?2b0e:3945
updateComponent @ vue.runtime.esm.js?2b0e:4066
get @ vue.runtime.esm.js?2b0e:4479
Watcher @ vue.runtime.esm.js?2b0e:4468
mountComponent @ vue.runtime.esm.js?2b0e:4073
Vue.$mount @ vue.runtime.esm.js?2b0e:8415
_callee$ @ main.ts?bc82:94
tryCatch @ runtime.js?96cf:45
invoke @ runtime.js?96cf:274
prototype.<computed> @ runtime.js?96cf:97
asyncGeneratorStep @ asyncToGenerator.js?1da1:3
_next @ asyncToGenerator.js?1da1:25
Promise.then (async)
asyncGeneratorStep @ asyncToGenerator.js?1da1:13
_next @ asyncToGenerator.js?1da1:25
```
No errors in the terminal (`electron:serve`).

**Comment by rathboma** [comment]
@UtechtDustin what version of node are you running?

**Comment by rathboma** [comment]
@geovannimp actually pretty worried about this IV-length error. Definitely something to do with the encryption. Seen this before?

**Comment by rathboma** [comment]
https://stackoverflow.com/questions/42485408/getting-error-of-invalid-iv-length-while-using-aes-256-cbc-for-encryption-in-nod <- seems to be the cause

**Comment by UtechtDustin** [comment]
> @UtechtDustin what version of node are you running?
v14.10.1

**Comment by rathboma** [comment]
I think I should just not be using the typeorm encryption plugin I was using here. I think this needs to be replaced with something like Simple Crypto

**Comment by rathboma** [comment]
Yeah. I'm going to revert that merge and re-implement using Simple-Encryption (which I'm already using successfully in a different context)

**Comment by rathboma** [comment]
Should be fixed in #335
